clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1633,45 +1633,54 @@ mlir::Value ScalarExprEmitter::buildScalarCast(
     DstElementType = DstType;
   }
 
-  if (CGF.getBuilder().isInt(SrcElementTy)) {
-    if (SrcElementType->isBooleanType() && Opts.TreatBooleanAsSigned) {
-      llvm_unreachable("NYI");
+  if (SrcElementTy.isa<mlir::IntegerType>() ||
+      DstElementTy.isa<mlir::IntegerType>())
+    llvm_unreachable("Obsolete code. Don't use mlir::IntegerType with CIR.");
+
+  if (SrcElementType->isBooleanType()) {
+    if (Opts.TreatBooleanAsSigned)
+      llvm_unreachable("NYI: signed bool");
+    if (CGF.getBuilder().isInt(DstElementTy)) {
+      return Builder.create<mlir::cir::CastOp>(
+          Src.getLoc(), DstTy, mlir::cir::CastKind::bool_to_int, Src);
+    } else if (DstTy.isa<mlir::FloatType>()) {
+      llvm_unreachable("NYI: bool->float cast");
+    } else {
+      llvm_unreachable("Unexpected destination type for scalar cast");
     }
-
-    if (CGF.getBuilder().isInt(DstElementTy))
+  } else if (CGF.getBuilder().isInt(SrcElementTy)) {
+    if (CGF.getBuilder().isInt(DstElementTy)) {
       return Builder.create<mlir::cir::CastOp>(
           Src.getLoc(), DstTy, mlir::cir::CastKind::integral, Src);
-    if (DstElementTy.isa<mlir::FloatType>())
+    } else if (DstElementTy.isa<mlir::FloatType>()) {
       return Builder.create<mlir::cir::CastOp>(
           Src.getLoc(), DstTy, mlir::cir::CastKind::int_to_float, Src);
-    llvm_unreachable("Unknown type cast");
-  }
-
-  // Leaving mlir::IntegerType around incase any old user lingers
-  if (DstElementTy.isa<mlir::IntegerType>()) {
-    llvm_unreachable("NYI");
-  }
-
-  if (DstElementTy.isa<mlir::cir::IntType>()) {
-    assert(SrcElementTy.isa<mlir::FloatType>() && "Unknown real conversion");
-
-    // If we can't recognize overflow as undefined behavior, assume that
-    // overflow saturates. This protects against normal optimizations if we are
-    // compiling with non-standard FP semantics.
-    if (!CGF.CGM.getCodeGenOpts().StrictFloatCastOverflow)
-      llvm_unreachable("NYI");
-
-    if (Builder.getIsFPConstrained())
-      llvm_unreachable("NYI");
-    return Builder.create<mlir::cir::CastOp>(
-        Src.getLoc(), DstTy, mlir::cir::CastKind::float_to_int, Src);
+    } else {
+      llvm_unreachable("Unexpected destination type for scalar cast");
+    }
+  } else if (SrcElementTy.isa<mlir::FloatType>()) {
+    if (CGF.getBuilder().isInt(DstElementTy)) {
+      // If wwe can't recognize overflow as undefined behavior, assume that
+      // overflow saturates.  This protects against normal optimizations if we
```

Comment:
nit: tide up `.  This` -> `. This`.

Reply:
I have always been a 2-spaces-between-sentences person.  I have no problem changing to one space when it is pointed out, but this is something I will likely do very frequently.

Reply:
Sounds good. The [LLVM coding guidelines](https://llvm.org/docs/CodingStandards.html) (which we try to follow) doesn't say anything about it and even tho I'm used to nit on this, it's not a problem to use 2-spaces-between-sentences!